api/server/secrets.go

```diff
@@ -0,0 +1,148 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/libopenstorage/openstorage/secrets"
+)
+
+const (
+	secretKeyOkMsg   = "Secret Key set successfully"
+	secretLoginOkMsg = "Secrets Login Succeeded"
+	secretLoginCheck = "Secrets Login Check Succeeded"
+)
+
+// TODO: Add swagger yaml
+func (c *clusterApi) setDefaultSecretKey(w http.ResponseWriter, r *http.Request) {
+
+	method := "setDefaultSecretKey"
+	var secReq secrets.DefaultSecretKeyRequest
+	var secResp secrets.SecretResponse
+
+	if err := json.NewDecoder(r.Body).Decode(&secReq); err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err := c.SecretManager.SetDefaultSecretKey(
+		secReq.DefaultSecretKey,
+		secReq.Override,
+	)
+
+	if err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	secResp.Status = "Default " + secretKeyOkMsg
```

Comment:
Unless this is the way it is supposed to be expected, if there is no returned data expected, a returned HTTP status of 200 is all that is needed. So line 38-39 may not be necessary. Just need to ensure we are returning 200.

Again, that is as long as no data is expected in return, because it looks like to me that line 38 is just returning constant message.

Reply:
@adityadani In my understanding no return data is expected for setting secrets, is should be ok if we return only HTTP status 200?

Reply:
I think set calls could just return an status code with empty response body. If we return anything it should be the set secret key-value, to be RESTful. But I think, we should go with just status code for now.